packages/web-frontend/src/components/SearchBarItem.js

```diff
@@ -22,80 +65,96 @@ const SearchBarItemComponent = ({
   organisationUnitCode,
   name,
   organisationUnitChildren,
-  isLoading,
   type,
   onClick,
+  onClickExpand,
   nestedMargin,
+  isFinalRow,
 }) => {
-  const [isExpanded, setIsExpanded] = useState(false);
+  const [isOpen, setIsOpen] = useState(false);
 
-  const nestedItems = isExpanded
-    ? organisationUnitChildren.map(child => (
-        <SearchBarItem key={child} organisationUnitCode={child} />
-      ))
-    : [];
-  return (
-    <HierarchyItem
-      key={organisationUnitCode}
-      label={name}
-      nestedMargin={nestedMargin}
-      nestedItems={nestedItems}
-      hasNestedItems={type === 'Country' || organisationUnitChildren.length > 0}
-      isLoading={isLoading}
-      Icon={ICON_BY_ORG_UNIT_TYPE[type]}
-      onClick={() => {
-        setIsExpanded(!isExpanded);
-        onClick(organisationUnitCode);
-      }}
+  const nestedItems = organisationUnitChildren.map((child, index) => (
+    <SearchBarItem
+      key={child}
+      organisationUnitCode={child}
+      onClick={onClick}
+      nestedMargin={nestedMargin + 24}
+      isFinalRow={isFinalRow && index === organisationUnitChildren.length - 1}
     />
+  ));
+  // always show an expander for country org units, which lazy load their children
+  const hasNestedItems = type === 'Country' || organisationUnitChildren.length > 0;
+
+  const Icon = ICON_BY_ORG_UNIT_TYPE[type];
+
+  return (
+    <Container key={organisationUnitCode}>
+      <Row $hideBottomBorder={isFinalRow && !isOpen}>
+        <StyledButton onClick={() => onClick(organisationUnitCode)} $nestedMargin={nestedMargin}>
+          {name}
+          {Icon && <Icon style={{ opacity: 0.7 }} />}
+        </StyledButton>
+        {hasNestedItems && (
+          <OpenCloseButton
```

Comment:
Yep there was previously a spinning icon on the desktop version, but actually there's since been an overall spinner that fills the entire screen whenever the entity hierarchy is loading, so it seemed unnecessary to have a small one in the menu as well (which was basically concealed by the semi-transparent overall spinner overlay)

Reply:
The lagging I refer to is when you open the entity hierarchy page and try to expand an entity (e.g. a country), it will take some time to render the sub entities search bar item.

Reply:
Emm I see the spinning icon for the country now, but not for district level.

Reply:
Good point, I thought you were talking about the lazy loading of country entities, but you mean the render lag further down the hierarchy - will take a look at it 😃

Reply:
@billli0 I spent some time looking into why it might take so long, and it seems there are a lot of inefficiencies in Tupaia - too many to deal with in this PR! Both around the handling of state and selectors, and the component hierarchy having rerenders cascade through unnecessarily

Reply:
Yeah there is some space to improve in tupaia frontend. This is more like a core issue in saga, but memo and callback can help a bit for now.